django_socio_grpc/tests/test_permissions.py

```diff
@@ -0,0 +1,204 @@
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_permissions_not_ok(self, mock_get_permissions, mock_permission_denied):
+        mock_get_permissions.return_value = [FakePermission()]
+        mock_get_permissions.return_value = [FakePermission()]
+        #   Create a dummyservice for unitary tests
+        dummy_service = DummyService()
+        dummy_service.context = FakeContext()
+        #   Call func
+        with mock.patch(
+            "django_socio_grpc.tests.test_permissions.FakePermission.has_permission"
+        ) as mock_has_permissions:
+            mock_has_permissions.return_value = False
+            dummy_service.check_permissions()
+            mock_has_permissions.assert_called_once_with(dummy_service.context, dummy_service)
+            mock_permission_denied.assert_called_once_with(
+                message="fake message", code="fake code"
+            )
+
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_object_permissions_ok(self, mock_get_permissions):
+        mock_get_permissions.return_value = [FakePermission()]
+        #   Create a dummyservice for unitary tests
+        dummy_service = DummyService()
+        dummy_service.context = FakeContext()
+        #   Call func
+        with mock.patch(
+            "django_socio_grpc.tests.test_permissions.FakePermission.has_object_permission"
+        ) as mock_has_object_permissions:
+            mock_has_object_permissions.return_value = True
+            dummy_service.check_object_permissions("fake_obj")
+            mock_has_object_permissions.assert_called_once_with(
+                dummy_service.context, dummy_service, "fake_obj"
+            )
+
+    @mock.patch("django_socio_grpc.services.Service.permission_denied")
+    @mock.patch("django_socio_grpc.services.Service.get_permissions")
+    def test_check_object_permissions_not_ok(
+        self, mock_get_permissions, mock_permission_denied
+    ):
+        mock_get_permissions.return_value = [FakePermission()]
+        #   Create a dummyservice for unitary tests
+        dummy_service = DummyService()
+        dummy_service.context = FakeContext()
+        #   Call func
+        with mock.patch(
+            "django_socio_grpc.tests.test_permissions.FakePermission.has_object_permission"
+        ) as mock_has_object_permissions:
+            mock_has_object_permissions.return_value = False
+            dummy_service.check_object_permissions("fake_obj")
+            mock_has_object_permissions.assert_called_once_with(
+                dummy_service.context, dummy_service, "fake_obj"
+            )
+            mock_permission_denied.assert_called_once_with(
+                message="fake message", code="fake code"
+            )
+
+    def test_get_permissions(self):
+        dummy_service = DummyService()
+        dummy_service.permission_classes = [FakePermission]
```

Comment:
in the other test FakePermission is called and here it is not why ? (FakePermission() -> FakePermission)

Reply:
self.get_permissions() return instance of permission objects and permission_classes attr is a list of permission classes